homeassistant/components/nextbus/__init__.py

```diff
@@ -27,7 +27,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     coordinator.add_stop_route(entry_stop, entry.data[CONF_ROUTE])
 
-    await coordinator.async_config_entry_first_refresh()
+    await coordinator.async_refresh()
+    if not coordinator.last_update_success:
+        return False
```

Comment:
This doesn't seem right, doesn't returning `False` here mean the config entry setup permanently fails, unlike if we raise `ConfigEntryNotReady`?

Reply:
> This doesn't seem right, doesn't returning `False` here mean the config entry setup permanently fails, unlike if we check for exceptions and raise `ConfigEntryNotReady`?

Ah indeed!
I have fixed this, and added specific tests